File: 1 - Data Preprocessing/data_preprocessing.r
# Importing the dataset
dataset = read.csv(file = "Data.csv")
#dataset = dataset[, 2:3] #

# Splitting the dataset in train and test set
library(caTools)
set.seed(123)
split = sample.split(Y = dataset$Purchased, SplitRatio = 0.8)
training_set = subset(dataset, split == TRUE)
testing_set = subset(x = dataset, subset = split == FALSE)

# Feature Scaling
#training_set[, 2:3] = scale(x = training_set[,2:3])
#testing_set[, 2:3] = scale(x = testing_set[,2:3])

File: 1 - Data Preprocessing/data_preprocessing.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Importing datasets
dataset = pd.read_csv("Data.csv")
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 3].values

#Handling missing data
from sklearn.preprocessing import Imputer
imputer = Imputer(missing_values='NaN', strategy='mean', axis=0)
imputer = imputer.fit(X[:,1:3]) #Fitting columns with missing values
X[:,1:3] = imputer.transform(X[:,1:3]) #Replace missing data on X

#Encoding categorical variables
from sklearn.preprocessing import LabelEncoder, OneHotEncoder
label_encoder_X = LabelEncoder()
X[:, 0] = label_encoder_X.fit_transform(X[:, 0])
ohe = OneHotEncoder(categorical_features=[0])
X = ohe.fit_transform(X).toarray()
label_encoder_Y = LabelEncoder()
y = label_encoder_Y.fit_transform(y)

#Splitting in training and testing sets
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

#Feature Scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test) #already fitted on training set